test/Microsoft.TestPlatform.AcceptanceTests/DataCollectionTests.cs

```diff
@@ -82,6 +82,19 @@ public void ExecuteTestsWithDataCollectionUsingCollectArgument(RunnerInfo runner
             this.VaildateDataCollectorOutput();
         }
 
+        [TestMethod]
+        [NetFullTargetFrameworkDataSource]
+        [NetCoreTargetFrameworkDataSource]
+        public void DataCollectorAssemblyLoadingShouldNotThrowError(RunnerInfo runnerInfo)
+        {
+            AcceptanceTestBase.SetTestEnvironment(this.testEnvironment, runnerInfo);
+
+            var arguments = PrepareArguments(GetAssetFullPath("TraceCollectorLoader.dll", "netcoreapp2.0"), string.Empty, string.Empty, this.FrameworkArgValue);
```

Comment:
Adding only netcoreapp2.0 since AppDomain is only available after netcoreapp2.0.
[Documentation](https://docs.microsoft.com/en-us/dotnet/api/system.appdomain.currentdomain?view=netframework-4.8)

Reply:
How does this run for net451 then ?

Reply:
And where is the datacollector getting enabled, I need to understand this better.

Reply:
We load the datacollectors only in the case of netcore as this change was made only in DotnetTestHostManager [here](https://github.com/microsoft/vstest/blob/master/src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs#L252). So net451 run without error.
` var allAssemblies = AppDomain.CurrentDomain.GetAssemblies();` in the test loads datacollectors including the trace datacollector.
When we attempt to get the types of the trace datacollector, in `var typeInfo = assembly.GetTypes(); `it breaks with the error.